estimagic/inference/bootstrap_samples.py

```diff
@@ -0,0 +1,86 @@
+import numpy as np
+
+from estimagic.inference.bootstrap_ci import _check_inputs
+
+
+def get_seeds(n_draws=1000):
+    """Draw seeds for bootstrap resampling.
+
+    Args:
+        n_draws (int): number of bootstrap draws.
+
+    Returns:
+        seeds (numpy.array): vector of randomly drawn seeds.
+
+    """
+
+    return np.random.randint(0, 2 ** 31, size=n_draws)
+
+
+def get_bootstrap_samples(
+    data, cluster_by=None, seeds=None, n_draws=1000, n_cores=1, return_samples=False
+):
+    """Draw and return bootstrap samples, either by specified seeds or number of draws.
+
+    Args:
+        data (pandas.DataFrame): original dataset.
+        cluster_by (str): column name of the variable to cluster by.
+        seeds (numpy.array): Size n_draws vector of drawn seeds or None.
+        n_draws (int): number of draws, only relevant if seeds is None.
+        n_cores (int): number of jobs for parallelization.
+        return_samples (bool): If true, return samples, else return indices.
+
+    Returns:
+        samples (list): list of DataFrames containing resampled data or ids.
+
+    """
+
+    _check_inputs(data=data, cluster_by=cluster_by)
+
+    if seeds is None:
+        seeds = get_seeds(n_draws)
+
+    if cluster_by is None:
+
+        from estimagic.inference.bootstrap_estimates import _get_uniform_estimates
```

Comment:
Imports should always be at the top of the module. Is there a reason to do it here? And private function should not be importable at all except in test modules.

Reply:
Reason were circular import problems.

Reply:
Done in #202